src/controller.ts

```diff
@@ -2,59 +2,53 @@ import * as vscode from 'vscode';
 import { dirname, join } from 'path';
 import { clearDecorations, setLinesHeat } from './view';
 import { absolutePath, AustinStats } from './model';
+import { getAustinCommand } from './utils/commandFactory';
 
 
 function delay(ms: number) {
     return new Promise(resolve => setTimeout(resolve, ms));
 }
 
-
 export class AustinController {
 
-    public constructor(private stats: AustinStats) { }
+    public constructor(private stats: AustinStats, private output: vscode.OutputChannel) { }
 
     public async profileScript() {
-        const pythonExtension = vscode.extensions.getExtension("ms-python.python");
```

Comment:
I'm slightly concerned about delaying the extension check to `getAustinCommand` til after we create a terminal which we might not even use if an exception is thrown before. Perhaps we can put the check in its own method for a bit of clean-up and reuse, but still call it on entry here?